packages/app/src/systems/Settings/hooks/useSettings.ts

```diff
@@ -0,0 +1,60 @@
+import { toast } from '@fuel-ui/react';
+import { useInterpret, useSelector } from '@xstate/react';
+import { useNavigate } from 'react-router-dom';
+
+import type { SettingsMachineState } from '../machines/settingsMachine';
+import { settingsMachine } from '../machines/settingsMachine';
+
+import type { AccountInputs } from '~/systems/Account';
+import { useAccount } from '~/systems/Account';
+import { Pages } from '~/systems/Core';
+
+const selectors = {
+  isChangingPassword: (state: SettingsMachineState) =>
+    state.matches('changingPassword'),
+  isUnlocking: (state: SettingsMachineState) => state.hasTag('unlocking'),
+  isGettingMnemonic: (state: SettingsMachineState) =>
+    state.matches('gettingMnemonic'),
+  words: (state: SettingsMachineState) => state.context.words,
+};
+
+export function useSettings() {
+  const navigate = useNavigate();
+  const { account } = useAccount();
+  const service = useInterpret(() =>
+    settingsMachine.withConfig({
+      actions: {
+        goToWallet() {
+          toast.success('Password Changed');
```

Comment:
`toast` action is not dependent on react, should be done inside the machine.

let's not separate logic and keep everything inside the machines. `navigate` is an exception because it comes from `useNavigate`, but we should also fix it to be inside the machine in the future